core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -1623,6 +1670,57 @@ public void registerNewWorkerConfListener(BiConsumer<Address, List<ConfigPropert
     mWorkerRegisteredListeners.add(function);
   }
 
+  /**
+   * Start decommission process for a specific worker.
+   */
+  private void startDecommission(MasterWorkerInfo worker) {
+    // TODO(bzheng) Handle the state and replicas in excluded workers.
+    try (LockResource r = worker.lockWorkerMeta(
+        EnumSet.of(WorkerMetaLockSection.BLOCKS), false)) {
+      processLostWorker(worker);
+    }
+  }
+
+  /**
+  * Decommission workers and update excluded worker map.
+  */
+  @Override
+  public void decommissionWorkers(final Set<String> excludedWorkerSet, boolean clear) {
+    Map<String, MasterWorkerInfo> tempWorkers = new HashMap<>();
+    if (excludedWorkerSet.isEmpty()) {
+      mDecommissionedWorkers.clear();
+      mDecommissioningWorkers.clear();
+      LOG.warn("Clear decommissioned worker map.");
```

Comment:
Why clearing the states for decommissioning workers? If I invoke the command with an empty list, does it make the master forget about all decommission-in-progress workers?

Reply:
Yes, this just provide a way to reactive the decommissioned workers, but the more 
 graceful way is to do this with a clear shell command.